### ISSUE TITLE
Rule to report superfluous default imports

### ISSUE THREAD

**Issue by sschuberth** [detekt/detekt#3699]
## Expected Behavior of the rule

Kotlin has a list of [default imports](https://kotlinlang.org/docs/packages.html#default-imports) which are superfluous to specify. *detekt* should be able to report them so they can be removed.

## Context

I was coming across the explicit use of default imports in a project that already uses *detekt*, and was surprised that *detekt* does not seem to have a rule for this yet. Also, such a rule should be enabled be default, IMO.

As an idea for for the implementation, maybe the default imports could become the default list for the existing [ForbiddenImport](https://detekt.github.io/detekt/style.html#forbiddenimport) rule.

**Comment by cortinico** [comment]
> maybe the default imports could become the default list for the existing ForbiddenImport rule.

I like this rule idea :) Just a small note, I would not do this change as `ForbiddenImport` should be used for imports that are considered prohibited (i.e. should be considered an error). Adding the default imports should be instead considered like a warning (i.e. it's not a mistake to specify them, is just not useful).

**Comment by DennisPietruck0001** [comment]
It seems that this issue can not be resolved since there is no way to identify whether an import statement is absolutely unnecessary. As described in this [comment](https://github.com/detekt/detekt/pull/3748#discussion_r628729903) there are cases where it is necessary to import a default imported package. If the declaration is in the same file it is possible to detect that the import is required.

But if the declaration is in another file of the same package we can not check whether the import is unnecessary.

Example:

```kotlin
// File1.kt
package test

fun println(){}
```

```kotlin
// File2.kt
package test

import kotlin.io.println // without this import the function declared in File1.kt is used

fun doPrint() {
  println()
}
```

Unless there is a way to know that there is a `println` defined in the same package it is impossible to tell whether the import statement in File2.kt is required or not.

**Comment by BraisGabin** [comment]
I assume that `BindingContext` could give us that information... But I'm not sure how.
